File: app/schemas/billing.py
from pydantic import BaseModel
from typing import Optional
from decimal import Decimal

class BillingCreate(BaseModel):
    appointment_id: int
    amount: Decimal
    status: str
    
class BillingResponse(BaseModel):
    id: int
    appointment_id: int
    amount: Decimal
    status: str
    
class BillingUpdate(BaseModel):
    amount: Optional[Decimal] = None
    status: Optional[str] = None

File: frontend/src/types/billing.ts
export interface Billing {
  id: number;
  appointment_id: number;
  amount: string;
  status: string;
}

export interface BillingCreate {
  appointment_id: number;
  amount: number;
  status: string;
}

export interface BillingUpdate {
  amount?: number;
  status?: string;
}